x-pack/plugins/apm/public/components/alerting/rule_types/transaction_duration_rule_type/index.tsx

```diff
@@ -41,6 +42,7 @@ export interface RuleParams {
   serviceName: string;
   threshold: number;
   transactionType: string;
+  transactionName: string;
```

Comment:
Can we make this string | undefined or string | null?

Reply:
since we support `All` option which means don't filter by the field I need to change it to 

```
export interface RuleParams {
  aggregationType: AggregationType;
  windowSize: number;
  windowUnit: string;
 threshold: number;
  environment: string; 
  serviceName?: string;
  transactionType?: string;
  transactionName?: string;
}

the service environment will remain as string since we need to pass  "environment": "ENVIRONMENT_ALL"
```